### ISSUE TITLE
Fibaro Roller Shutter 2 multiple MQTT

### ISSUE THREAD

**Issue by jonferreira** [domoticz/domoticz#2500]
Been living with this issue since I made the move to Domoticz, but only nowit has become really annoying given that I can't use any of these devices as a "scene trigger"

Have a couple of FIBARO System FGRM222 Roller Shutter Controller 2 which work "fine" except for the fact when once controls them via Z-Wave it sends 2 "spurious" MQTT message.

How to reproduce:

Have a FIBARO System FGRM222 Roller Shutter Controller 2 set to 85%.
On domoticz UI change level to 60%

What was expected:

A single MQTT message such as 

`{"topic":"domoticz/out","payload":{"Battery":255,"RSSI":12,"description":"","dtype":"Light/Switch","id":"00001501","idx":190,"name":"Office Roller Shutter","nvalue":2,"stype":"Switch","svalue1":"60","switchType":"Blinds Percentage Inverted","unit":1},"qos":0,"retain":false,"_topic":"domoticz/out","_msgid":"5c1a63e4.c89ecc"}
`
What we get:

Multiple MQTT messages such as:

```
{"topic":"domoticz/out","payload":{"Battery":255,"RSSI":12,"description":"","dtype":"Light/Switch","id":"00001501","idx":190,"name":"Office Roller Shutter","nvalue":2,"stype":"Switch","svalue1":"60","switchType":"Blinds Percentage Inverted","unit":1},"qos":0,"retain":false,"_topic":"domoticz/out","_msgid":"518e2374.76ea5c"}

{"topic":"domoticz/out","payload":{"Battery":255,"RSSI":12,"description":"","dtype":"Light/Switch","id":"00001501","idx":190,"name":"Office Roller Shutter","nvalue":2,"stype":"Switch","svalue1":"85","switchType":"Blinds Percentage Inverted","unit":1},"qos":0,"retain":false,"_topic":"domoticz/out","_msgid":"61fe6b31.93e2e4"}

{"topic":"domoticz/out","payload":{"Battery":255,"RSSI":12,"description":"","dtype":"Light/Switch","id":"00001501","idx":190,"name":"Office Roller Shutter","nvalue":2,"stype":"Switch","svalue1":"60","switchType":"Blinds Percentage Inverted","unit":1},"qos":0,"retain":false,"_topic":"domoticz/out","_msgid":"5c1a63e4.c89ecc"}

```
Basically we get a first notification with the "set value" (60%) but then it seens we immediatly get another message with the "current value" (85%) and then later again another message with 60%, probably due to polling or the module ACK

Hope this makes sense

**Comment by gizmocuz** [comment]
I think this is coming from the node itself (and openzwave)
You could test this by enabling the OZW debug log (from the hardware setup page)
Then restart the hardware/domoticz, wait until all nodes are queried and then control the blind
Then have a look in the log
Disable logging as soon as possible as it will become a very large file
If that is the case, then it will be very hard to 'fix' this, as currently we parse all messages to MQTT

**Comment by jonferreira** [comment]
morning @gizmocuz 

noob question: where would this OZW log be stored at?

**Comment by gizmocuz** [comment]
@jonferreira, in the config folder ozw_xxxxxx.log (or simular)

**Comment by jonferreira** [comment]
ah found it thanks! will reply shortly

**Comment by jonferreira** [comment]
```
2018-06-21 09:11:39.057 Info, Node021, Value::Set - COMMAND_CLASS_SWITCH_MULTILEVEL - Level - 0 - 1 - 70
2018-06-21 09:11:39.058 Info, Node021, SwitchMultilevel::Set - Setting to level 70
2018-06-21 09:11:39.059 Info, Node021,   Duration: Default
2018-06-21 09:11:39.060 Detail, Node021, Queuing (Send) SwitchMultilevelCmd_Set (Node=21): 0x01, 0x0b, 0x00, 0x13, 0x15, 0x04, 0x26, 0x01, 0x46, 0xff, 0x25, 0xf4, 0xb9
2018-06-21 09:11:39.061 Detail, Node021, Queuing (Send) SwitchMultilevelCmd_Get (Node=21): 0x01, 0x09, 0x00, 0x13, 0x15, 0x02, 0x26, 0x02, 0x25, 0xf5, 0x06
2018-06-21 09:11:39.062 Detail, 
2018-06-21 09:11:39.062 Info, Node021, Sending (Send) message (Callback ID=0xf4, Expected Reply=0x13) - SwitchMultilevelCmd_Set (Node=21): 0x01, 0x0b, 0x00, 0x13, 0x15, 0x04, 0x26, 0x01, 0x46, 0xff, 0x25, 0xf4, 0xb9
2018-06-21 09:11:39.071 Detail, Node021,   Received: 0x01, 0x04, 0x01, 0x13, 0x01, 0xe8
2018-06-21 09:11:39.072 Detail, Node021,   ZW_SEND_DATA delivered to Z-Wave stack
2018-06-21 09:11:39.089 Detail, Node021,   Received: 0x01, 0x07, 0x00, 0x13, 0xf4, 0x00, 0x00, 0x02, 0x1d
2018-06-21 09:11:39.089 Detail, Node021,   ZW_SEND_DATA Request with callback ID 0xf4 received (expected 0xf4)
2018-06-21 09:11:39.090 Info, Node021, Request RTT 27 Average Request RTT 26
2018-06-21 09:11:39.091 Detail,   Expected callbackId was received
2018-06-21 09:11:39.092 Detail,   Expected reply was received
2018-06-21 09:11:39.093 Detail,   Message transaction complete
2018-06-21 09:11:39.094 Detail, 
2018-06-21 09:11:39.094 Detail, Node021, Removing current message
2018-06-21 09:11:39.095 Detail, 
2018-06-21 09:11:39.095 Info, Node021, Sending (Send) message (Callback ID=0xf5, Expected Reply=0x04) - SwitchMultilevelCmd_Get (Node=21): 0x01, 0x09, 0x00, 0x13, 0x15, 0x02, 0x26, 0x02, 0x25, 0xf5, 0x06
2018-06-21 09:11:39.105 Detail, Node021,   Received: 0x01, 0x04, 0x01, 0x13, 0x01, 0xe8
2018-06-21 09:11:39.105 Detail, Node021,   ZW_SEND_DATA delivered to Z-Wave stack
2018-06-21 09:11:39.122 Detail, Node021,   Received: 0x01, 0x07, 0x00, 0x13, 0xf5, 0x00, 0x00, 0x02, 0x1c
2018-06-21 09:11:39.122 Detail, Node021,   ZW_SEND_DATA Request with callback ID 0xf5 received (expected 0xf5)
2018-06-21 09:11:39.122 Info, Node021, Request RTT 25 Average Request RTT 25
2018-06-21 09:11:39.123 Detail,   Expected callbackId was received
2018-06-21 09:11:39.132 Detail, Node021,   Received: 0x01, 0x09, 0x00, 0x04, 0x00, 0x15, 0x03, 0x26, 0x03, 0x32, 0xf3
2018-06-21 09:11:39.132 Detail, 
2018-06-21 09:11:39.133 Info, Node021, Response RTT 36 Average Response RTT 36
2018-06-21 09:11:39.133 Info, Node021, Received SwitchMultiLevel report: level=50
2018-06-21 09:11:39.133 Detail, Node021, Refreshed Value: old value=50, new value=50, type=byte
2018-06-21 09:11:39.134 Detail, Node021, Changes to this value are not verified
2018-06-21 09:11:39.134 Detail, Node021,   Expected reply and command class was received
2018-06-21 09:11:39.135 Detail, Node021,   Message transaction complete
2018-06-21 09:11:39.135 Detail, 
2018-06-21 09:11:39.135 Detail, Node021, Removing current message
2018-06-21 09:11:39.136 Detail, Node021, Notification: ValueChanged
2018-06-21 09:11:40.249 Detail, Node021,   Received: 0x01, 0x0c, 0x00, 0x04, 0x00, 0x15, 0x06, 0x31, 0x05, 0x04, 0x22, 0x05, 0x55, 0xa6
2018-06-21 09:11:40.249 Detail, 
2018-06-21 09:11:40.249 Info, Node021, Received SensorMultiLevel report from node 21, instance 1, Power: value=136.5W
2018-06-21 09:11:40.250 Detail, Node021, Refreshed Value: old value=0.0, new value=136.5, type=decimal
2018-06-21 09:11:40.250 Detail, Node021, Changes to this value are not verified
2018-06-21 09:11:40.251 Detail, Node021, Notification: ValueChanged
2018-06-21 09:11:45.305 Detail, Node021,   Received: 0x01, 0x09, 0x00, 0x04, 0x00, 0x15, 0x03, 0x26, 0x03, 0x46, 0x87
2018-06-21 09:11:45.306 Detail, 
2018-06-21 09:11:45.306 Info, Node021, Received SwitchMultiLevel report: level=70
2018-06-21 09:11:45.306 Detail, Node021, Refreshed Value: old value=50, new value=70, type=byte
2018-06-21 09:11:45.306 Detail, Node021, Changes to this value are not verified
2018-06-21 09:11:45.306 Detail, Node021, Notification: ValueChanged
2018-06-21 09:11:45.325 Detail, Node021,   Received: 0x01, 0x0c, 0x00, 0x04, 0x00, 0x15, 0x06, 0x31, 0x05, 0x04, 0x22, 0x00, 0x00, 0xf6
2018-06-21 09:11:45.325 Detail, 
2018-06-21 09:11:45.325 Info, Node021, Received SensorMultiLevel report from node 21, instance 1, Power: value=0.0W
2018-06-21 09:11:45.326 Detail, Node021, Refreshed Value: old value=136.5, new value=0.0, type=decimal
2018-06-21 09:11:45.326 Detail, Node021, Changes to this value are not verified
2018-06-21 09:11:45.326 Detail, Node021, Notification: ValueChanged

```

This is what I got

**Comment by jonferreira** [comment]
So this clearly seems like a proble at the OZW level, not domoticz.

You can see I initially set the blind to 70%

2018-06-21 09:11:39.058 Info, Node021, SwitchMultilevel::Set - Setting to level 70

Then half way received some sort of spurious report with the "old" value

2018-06-21 09:11:39.133 Info, Node021, Received SwitchMultiLevel report: level=50
2018-06-21 09:11:39.133 Detail, Node021, Refreshed Value: old value=50, new value=50, type=byte

And the ACK at the end

2018-06-21 09:11:45.306 Info, Node021, Received SwitchMultiLevel report: level=70
2018-06-21 09:11:45.306 Detail, Node021, Refreshed Value: old value=50, new value=70, type=byte

Presumably it should be possible to suppress the "second" message given that old value = new value but that would still leave us with the ACK.

Wondering why this only happens with the Fibaro RS modules and not with "every dimmer" module if that makes sense

**Comment by gizmocuz** [comment]
It is probably also not a problem with OZW but with the node itself
Best to ask fibaro first, or google ?

**Comment by jonferreira** [comment]
Fibaro doesn't acknowledges any problems at all unless you use their controllers and even so, you'd be on for a long long discussion... 

Obviously it's something specifical to these modules and could even be fw specific, just find it odd that I've never noticed it while using Vera

**Comment by gizmocuz** [comment]
@jonferreira , could you check the groups the fibaro is using ? Did you add the controller only to the life-line (most times group 1), or did you add the control to multiple groups ?
(According the manual, it is group 3 (III association group ))

**Comment by jonferreira** [comment]
Yes lifeline group (3) only :)

Looking at the logs it really seems it's OZW who's sending a "spurious" GET for some reason

**Comment by jonferreira** [comment]
@gizmocuz wondering if you know what to do with the reply at https://github.com/OpenZWave/open-zwave/issues/1541

**Comment by waltervl** [comment]
This issue can be closed, OZW issue, not Domoticz.
